### ISSUE TITLE
Future steps of ClimatePlots

### ISSUE THREAD

**Issue by Balinus** [JuliaClimate/ClimatePlots.jl#9]
Here's a discussion about the future of `ClimatePlots`.

The initial state of this package is a split of what was available in [`ClimateTools`](https://github.com/Balinus/ClimateTools.jl). It mainly serves the type `ClimGrid` defined in ClimateTools (an in-memory representation of a netCDF dataset). At the time, plotting capabilities of georeferenced fields were not mature in Julia and the solution was to use Basemap under-the-hood. 

For ClimatePlots 1.0, I propose we explore, scout and list possible solutions for a more generic plotting framework.

[ESDLPlots.jl](https://github.com/esa-esdl/ESDLPlots.jl) Here's a library that have some nice ideas and implemented features. This is the plotting facility of [ESDL.jl](https://github.com/esa-esdl/ESDL.jl). 

[Maps.jl](https://github.com/NHDaly/Maps.jl). Geographical plotting using Makie. Would be useful for animations!

**Comment by Balinus** [comment]
Version 0.2 now uses cartopy under-the-hood. Right now I consider it to be rather non-flexible though, since the ``mapclimgrid` function is a do-it-all function than tries to do everything. Maintenance is thus harder. 

The short-term plan is to create a nice wrapper of cartopy commands

e.g.
```
ClimatePlots.contourf(C::ClimGrid; kwargs...)
ClimatePlots.pcolormesh(C::ClimGrid; kwargs...)
...
```

**Comment by Datseris** [comment]
Slowly by slowly I have been managing to plot my fields with Makie, and produce a vide of them. Doing the video is around 100 to 1000 times faster with Makie. 

The time gained in plotting performance is lost in debugging, and asking the developers "how this can be done", then finding out that it can't be done, then waiting for it to be implemented. But we are slowly getting there. 

I'll make a small code snippet example and post it here. Are you interested in GeoMakie?

**Comment by gaelforget** [comment]
Not sure how much this helps but:

- I also took a look at `Makie` (& `GeoMakie`) recently, which lead to the addition of `src/plot_makie.jl` and `examples/LatLonCap300mDepth.png` in https://github.com/JuliaClimate/IndividualDisplacements.jl 
- my earlier `examples/ex_1_mitgcm.png` was created using pyplot and cartopy (`src/plot_pyplot.jl`)

**Comment by Balinus** [comment]
@Datseris Yes, I think that it is quite useful to have some code that would allow to create some generic animation. GeoMakie seems a good candidate for our purpose. How much mature it is? I've seen you posted some Issues and got useful feedback from the developer(s).

@gaelforget Really nice images! This does provide some code that would certainly be useful for winds variables.
